### ISSUE TITLE
[java] UnusedModifier doesn't check static keyword on nested enum declaration

### ISSUE THREAD

**Issue by jsotuyod** [pmd/pmd#248]
**Rule Set:** [unused/UnusedModifier](https://pmd.github.io/pmd-5.5.3/pmd-java/rules/java/unusedcode.html#UnusedModifier)

**Description:**
Enums are `static` by default. Explicitly setting them matches this rule's criteria:

> For historical reasons, modifiers which are implied by the context are accepted by the compiler, but are superfluous.

**Code Sample demonstrating the issue:**

```java
public @interface Test {
  public static enum EnumSample {
    TEST;
  }
}
```

PMD will find no violations on this annotation, 1 violation is expected.

The behavior is consistent when nested in classes, annotations and interfaces.

**Running PMD through:** Any
